app/src/main/java/com/randomappsinc/studentpicker/importdata/ImportFromFileActivity.java

```diff
@@ -119,23 +94,23 @@ private void loadUI(String listNameText, String namesListText) {
     }
 
     @OnClick(R.id.save)
-    public void importNameList() {
+    public void importNameList(View view) {
         String newListName = listName.getText().toString().trim();
         if (newListName.isEmpty()) {
             UIUtils.showLongToast(R.string.blank_list_name, this);
         } else {
-            DataSource dataSource = new DataSource(this);
-            ListDO newList = dataSource.addNameList(newListName);
-
-            String[] allNames = names.getText().toString().split("\\r?\\n");
-            for (String name : allNames) {
-                String cleanName = name.trim();
-                if (!cleanName.isEmpty()) {
-                    dataSource.addNames(cleanName, 1, newList.getId());
-                }
-            }
+            loading.setVisibility(View.VISIBLE);
+            view.setEnabled(false);
```

Comment:
Disabling the view leads to some weird cases. For example, what if the saving fails; you need to re-enable the button. I also think it's weird just having a progress bar go over the UI. What if we use a blocking MaterialDialog instead? You can do indeterminate progress dialog with it and make it non-cancelable.

Reply:
yes, but what if the user click save many times? also progress dialog is deprecated but it's the better solution right now.

Reply:
There's a limit to how much we should protect the UI. I feel like the user double-tapping the button before the dialog shows up is incredibly difficult. If the user wants to break the app, let them break the app.